dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java

```diff
@@ -147,16 +150,21 @@ public Host resolveHostName(String serverName, User user, boolean respectFronten
     public Optional<Host> resolveHostNameWithoutDefault(String serverName, User user, boolean respectFrontendRoles) throws DotDataException, DotSecurityException {
 
         Host host = hostCache.getHostByAlias(serverName);
-        User systemUser = APILocator.systemUser();
-
-        if(host == null){
+        if (host != null && UtilMethods.isSet(host.getIdentifier())) {
```

Comment:
You can do this is one call - this will catch if the host is null too:

`if(UtilMethods.isSet(()->host.getIdentifier())){`

or 

`if(UtilMethods.isEmpty(()->host.getIdentifier())){`